module/x/gravity/types/ethereum.go

```diff
@@ -21,8 +23,81 @@ const (
 
 	// GravityDenomLen is the length of the denoms generated by the gravity module
 	GravityDenomLen = len(GravityDenomPrefix) + len(GravityDenomSeparator) + EthereumContractAddressLen
+
+	// ZeroAddressString ZeroAddress is an EthAddress containing the zero ethereum address
+	ZeroAddressString = "0x0000000000000000000000000000000000000000"
 )
 
+// EthAddress Regular EthAddress
+type EthAddress struct {
+	address common.Address
+}
+
+// GetAddress Returns the contained address as a string
+func (ea EthAddress) GetAddress() common.Address {
+	return ea.address
+}
+
+// SetAddress Sets the contained address, performing validation before updating the value
+func (ea *EthAddress) SetAddress(address string) error {
+	if err := ValidateEthAddress(address); err != nil {
+		return err
+	}
+	ea.address = common.HexToAddress(address)
+	return nil
+}
+
+func NewEthAddressFromBytes(address []byte) (*EthAddress, error) {
+	if err := ValidateEthAddress(hex.EncodeToString(address)); err != nil {
+		return nil, sdkerrors.Wrap(err, "invalid input address")
+	}
+
+	addr := EthAddress{common.BytesToAddress(address)}
+	return &addr, nil
+}
+
+// NewEthAddress Creates a new EthAddress from a string, performing validation and returning any validation errors
+func NewEthAddress(address string) (*EthAddress, error) {
+	if err := ValidateEthAddress(address); err != nil {
+		return nil, sdkerrors.Wrap(err, "invalid input address")
+	}
+
+	addr := EthAddress{common.HexToAddress(address)}
+	return &addr, nil
+}
+
+func has0xPrefix(str string) bool {
+	return len(str) >= 2 && str[0] == '0' && (str[1] == 'x' || str[1] == 'X')
+}
+
+// ValidateEthAddress Validates the input string as an Ethereum Address
+// Address must not be empty, have 42 character length, start with 0x and have 40 remaining characters in [0-9a-fA-F]
+func ValidateEthAddress(address string) error {
+	if address == "" {
+		return fmt.Errorf("empty")
+	}
+
+	// An auditor recommended we should check the error of hex.DecodeString, given that geth's HexToAddress ignores it
+	if has0xPrefix(address) {
+		address = address[2:]
+	}
+
+	if _, err := hex.DecodeString(address); err != nil {
+		return fmt.Errorf("invalid hex with error: %s", err)
+	}
```

Comment:
I guess this comment is expected to be placed in the next condition?
```suggestion
	if has0xPrefix(address) {
		address = address[2:]
	}

	// An auditor recommended we should check the error of hex.DecodeString, given that geth's HexToAddress ignores it
	if _, err := hex.DecodeString(address); err != nil {
		return fmt.Errorf("invalid hex with error: %s", err)
	}
```